libcipher/transpos.py

```diff
@@ -0,0 +1,92 @@
+"""Transposition cipher implementation
+
+Provides functions to encrypt/decrypt a message by transpose ciphering it.
+Encryption needs a key(a number).
+"""
+
+
+class Error(Exception):
+    pass
+
+
+class InvalidKeySizeException(Error):
+    '''Class created for throwing excption when invalid key size is given.
+
+    Attributes:
+        keysize (int): Size of Key
+        messagesize(int): Size of message
+
+    Args:
+        keysize (int): Size of Key
+        messagesize(int): Size of message
+    '''
+
+    def __init__(self, keysize, messagesize):
+        self.keysize = keysize
+        self.messagesize = messagesize
+
+    def __str__(self):
+        return(
+            "Cipher key is limited to half the length of the message" +
+            " size.\nKey Size is " + repr(self.keysize) +
+            " and Message Size is " + repr(self.messagesize))
+
+
+class NoKeyGivenException(Error):
+    '''Class created for throwing exception when no key is given.
+    '''
+    def __init__(self):
+        pass
+
+    def __str__(self):
+        return('Key is needed for encryption')
+
+
+class NoMessageGivenException(Error):
+    '''Class created for throwing exception no message is given
+    '''
+    def __init__(self):
+        pass
+
+    def __str__(self):
+        return('Message is needed for encryption')
+
+
+def encrypt(keysize, message):
+    '''
+    Args :
+        Transposition Cipher Key Size,Message
+    Returns:
+        message encrypted using transposition cipher.'|' is added at to
+        indicate end of message.For invalid key size,InvalidKeySizeException
+        is thrown.For empty input message,'|' is returned.
+    '''
+    if message is None:
+        raise NoMessageGivenException
+    else:
+        message_size = len(message)
+    # the transposition cipher key is limited to half the length of the
+    # message it is used to encrypt
+    if keysize is None:
+        raise NoKeyGivenException
+    else:
+        if keysize <= message_size//2:
```

Comment:
You might want to try reading http://stackoverflow.com/questions/19684434/best-way-to-check-function-arguments-in-python